api/src/main/java/org/openmrs/module/locationbasedaccess/PatientSearchAdvise.java

```diff
@@ -0,0 +1,92 @@
+ * <p>
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs.module.locationbasedaccess;
+
+
+import org.aopalliance.aop.Advice;
+import org.aopalliance.intercept.MethodInvocation;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.Location;
+import org.openmrs.Patient;
+import org.openmrs.Person;
+import org.openmrs.PersonAttribute;
+import org.openmrs.api.context.Context;
+import org.springframework.aop.Advisor;
+import org.springframework.aop.support.StaticMethodMatcherPointcutAdvisor;
+import java.lang.reflect.Method;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * It will be used to implement the AOP Methods for PatientServiceImpl class
+ */
+public class PatientSearchAdvise extends StaticMethodMatcherPointcutAdvisor implements Advisor {
+
+    protected static final Log log = LogFactory.getLog(PatientSearchAdvise.class);
+
+    @Override
+    public boolean matches(Method method, Class targetClass) {
+        if (method.getName().equals("getPatients")) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public Advice getAdvice() {
+        return new PatientSearchControlAdvice();
+    }
+
+    /**
+     * This inner class will be used to intercept the caught method using AOP
+     */
+    private class PatientSearchControlAdvice implements MethodInterceptor {
+
+        public Object invoke(MethodInvocation invocation) throws Throwable {
+            Object currentResult = invocation.proceed();
+            Integer sessionLocationId = Context.getUserContext().getLocationId();
+
+            if(sessionLocationId != null) {
+                Location sessionLocation = Context.getLocationService().getLocation(sessionLocationId);
+                List<Patient> currentPatientResults = (List<Patient>)currentResult;
+                for (Iterator<Patient> iterator = currentPatientResults.iterator(); iterator.hasNext(); ) {
+                    Patient patient  = iterator.next();
+                    patient = checkPatientForAccessLocation(patient, sessionLocation);
+                    if(patient == null) {
+                        iterator.remove();
+                    }
+                }
+                return currentPatientResults;
+            }
+            else {
+                // If session location doesn't available, return empty patient
+                log.debug("Search Patient : Null Session Location in the UserContext");
+                return  null;
+            }
+        }
+
+        private  Patient checkPatientForAccessLocation(Patient patient, Location sessionLocation) {
+            Person person = patient.getPerson();
+            final PersonAttribute personAttribute = person.getAttribute("accessControlLocation");
```

Comment:
What happens when the attribute name is not like what you have assumed above?

Reply:
It will return null value, and the patient will be excluded from the access Location filter. Simply user can get those patients who haven't location attribute into all locations without considering access location filter.